src/main/java/com/cst438/controller/AssignmentController.java

```diff
@@ -17,102 +16,214 @@
-
         // get the list of enrollments for the section related to this assignment.
-		// hint: use te enrollment repository method findEnrollmentsBySectionOrderByStudentName.
+        // hint: use te enrollment repository method findEnrollmentsBySectionOrderByStudentName.
         // for each enrollment, get the grade related to the assignment and enrollment
-		//   hint: use the gradeRepository findByEnrollmentIdAndAssignmentId method.
-        //   if the grade does not exist, create a grade entity and set the score to NULL
+        //   hint: use the gradeRepository findByEnrollmentIdAndAssignmentId method.
+        // TODO  if the grade does not exist, create a grade entity and set the score to NULL
         //   and then save the new entity
-
-        return null;
-    }
-
-    // instructor uploads grades for assignment
-    // user must be instructor for the section
-    @PutMapping("/grades")
-    public void updateGrades(@RequestBody List<GradeDTO> dlist) {
-
-        // TODO
-
-        // for each grade in the GradeDTO list, retrieve the grade entity
-        // update the score and save the entity
-
-    }
-
-
-
-    // student lists their assignments/grades for an enrollment ordered by due date
-    // student must be enrolled in the section
-    @GetMapping("/assignments")
-    public List<AssignmentStudentDTO> getStudentAssignments(
-            @RequestParam("studentId") int studentId,
-            @RequestParam("year") int year,
-            @RequestParam("semester") String semester) {
-
-        // TODO remove the following line when done
-
-        // return a list of assignments and (if they exist) the assignment grade
-        //  for all sections that the student is enrolled for the given year and semester
-		//  hint: use the assignment repository method findByStudentIdAndYearAndSemesterOrderByDueDate
-
-        return null;
-    }
-}
+        //TEST URL http://localhost:8080/assignments/1/grades
+        @GetMapping("/assignments/{assignmentId}/grades")
+        public List<GradeDTO> getAssignmentGrades ( @PathVariable("assignmentId") int assignmentId){
+
+//        int sectionNo = assignmentRepository.findSectionNoByAssignmentId(assignmentId);
+            Assignment a = assignmentRepository.findById(assignmentId).orElse(null);
+            if(a==null){
+                throw  new ResponseStatusException( HttpStatus.NOT_FOUND, "not found");
+            }
+
+            List<Enrollment> enrollments = enrollmentRepository.findEnrollmentsBySectionNoOrderByStudentName(a.getSection().getSectionNo());
+            List<GradeDTO> assignmentGrades = new ArrayList<>();
+
+            for (Enrollment enrollment : enrollments) {
+                Grade grade = gradeRepository.findByEnrollmentIdAndAssignmentId(enrollment.getEnrollmentId(), assignmentId);
+
+                if (grade == null) {
+//                    grade = new Grade();
+//                    grade.setGradeId(grade.getGradeId());
```

Comment:
Shouldn't return null here. I think you want to add a grade DTO  with a null grade value?